### PR TITLE
Add v2 cluster status update

### PR DESCRIPTION
**Pull request by croomes**
For v2 clusters, marks node as ready once it accepts a connection on its api port.

```
Status:
  Members:
    Ready:
      10.61.0.145
      10.61.0.147
      10.61.0.146
  Nodes:
    10.61.0.145
    10.61.0.147
    10.61.0.146
  Phase:  Running
  Ready:  3/3
Events:
  Type     Reason         Age    From                       Message
  ----     ------         ----   ----                       -------
  Warning  ChangedStatus  3m27s  storageoscluster-operator  0/3 StorageOS nodes are functional
  Warning  ChangedStatus  37s    storageoscluster-operator  1/3 StorageOS nodes are functional
  Warning  ChangedStatus  34s    storageoscluster-operator  2/3 StorageOS nodes are functional
  Normal   ChangedStatus  3s     storageoscluster-operator  3/3 StorageOS nodes are functional. Cluster healthy
```

### CODE REVIEW

File: pkg/storageos/update_status.go [hunk 1]
```diff
@@ -97,6 +142,18 @@ func isHealthy(health *storageosv1.NodeHealth) bool {
 	return false
 }
 
+func isListening(host string, port string, timeout time.Duration) bool {
+	conn, err := net.DialTimeout("tcp", net.JoinHostPort(host, port), timeout)
+	if err != nil {
+		fmt.Println("Connecting error:", err)
```
Comment (darkowlzz) [inline pkg/storageos/update_status.go]: We should use the logger if this needs to be logged.

File: pkg/storageos/update_status.go [hunk 2]
```diff
@@ -57,9 +57,54 @@ func (s *Deployment) getStorageOSStatus() (*storageosv1.StorageOSClusterStatus,
...
+//
+// Note: NodeHealthStatus is deprecated and not set for v2.
+func (s *Deployment) getStorageOSV2Status(nodeIPs []string) (*storageosv1.StorageOSClusterStatus, error) {
+	var readyNodes int
+
+	totalNodes := len(nodeIPs)
+	memberStatus := new(storageosv1.MembersStatus)
+
+	for _, ip := range nodeIPs {
+		if isListening(ip, storageosapi.DefaultPort, 2*time.Second) {
```
Comment (darkowlzz) [inline pkg/storageos/update_status.go]: Let's use a constant time for the timeout passed here and in the v1 `getNodeHealth()`.

File: pkg/storageos/update_status.go (hunk 1 above)
Comment (croomes) [reply pkg/storageos/update_status.go]: It doesn't need to be logged - removing

File: pkg/storageos/update_status.go (hunk 2 above)
Comment (croomes) [reply pkg/storageos/update_status.go]: yes, done

**Pull request merged**